crates/telio-firewall/src/firewall.rs

```diff
@@ -685,6 +685,27 @@ impl StatefullFirewall {
                     || it == v6::EchoReply.0
                 {
                     it
+                } else if (it == v4::DestinationUnreachable.0
+                    || it == v4::TimeExceeded.0
+                    || it == v4::ParameterProblem.0)
+                    && ip.get_next_level_protocol() == IpNextHeaderProtocols::Icmp
+                {
+                    let packet = unwrap_option_or_return!(
+                        Ipv4Packet::try_from(&icmp_packet.payload()[4..]),
```

Comment:
we should avoid using `[]` operator here (wonder why linter is not crying bout it, because it will panic if `payload` is shorter than 4 bytes, which effectively could create potential DoS.